File: languages/go/client/entity/signer.go
package entity

import "github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"

type Signer interface {
	ToProto() *proto.Signer
}

type EcdsaSigner struct {
	Alg  proto.SignerAlg
	Args SignerArgs
}

func NewEcdsaSigner(args SignerArgs) EcdsaSigner {
	return EcdsaSigner{
		Alg: proto.SignerAlg_ES256K,
		Args: SignerArgs{
			PrivateKey: args.PrivateKey,
			CommonName: args.CommonName,
		},
	}
}

func (s EcdsaSigner) ToProto() *proto.Signer {
	return &proto.Signer{
		Alg:  s.Alg,
		Args: s.Args.ToProto(),
	}
}

type SignerArgs struct {
	PrivateKey string
	CommonName *string
}

func (s SignerArgs) ToProto() *proto.SignerArgs {
	return &proto.SignerArgs{
		PrivateKey: &s.PrivateKey,
		CommonName: s.CommonName,
	}
}

File: languages/python/bloock/client/entity/signer.py
from abc import abstractmethod
import bloock._bridge.proto.record_pb2 as proto


class SignerArgs:
    def __init__(self, private_key: str, common_name=None) -> None:
        self.private_key = private_key
        self.common_name = common_name

    def to_proto(self) -> proto.SignerArgs:
        return proto.SignerArgs(
            private_key=self.private_key, common_name=self.common_name
        )


class Signer:
    def __init__(self, alg: proto.SignerAlg.ValueType, args: SignerArgs) -> None:
        self.alg = alg
        self.args = args

    @abstractmethod
    def to_proto(self):
        raise NotImplementedError


class EcdsaSigner(Signer):
    def __init__(self, private_key: str, common_name=None) -> None:
        super().__init__(alg=proto.ES256K, args=SignerArgs(private_key, common_name))

    def to_proto(self) -> proto.Signer:
        return proto.Signer(alg=self.alg, args=self.args.to_proto())
